mitodl/open-discussions
# upgrade linting and formatting tooling

#### What are the relevant tickets?

none, just keeping some dependencies up to date

#### What's this PR do?

I updated `babel-eslint`, `eslint`, `eslint-plugin-babel`, `eslint-plugin-flowtype`, `eslint-plugin-mocha`, `eslint-plugin-react`, and `prettier-eslint-cli` to their latest version. There are some related changes:

- prettier now formats a few things differently, in particular ternaries with JSX. it now formats them like this:

```js
{fooBar ? (
  <div>BOO</div>
) : null}
```

instead of

```js
{ fooBar
  ? <div>BOO</div>
  : null}
```

which is kind of nice I think, but did change how a lot of things were formatted.

- I also had to add a setting for the `eslint-plugin-react` package to tell it what react version we're on.


#### How should this be manually tested?

Well, this is just linting and formatting stuff, so nothing should be different! I would click around the app and make sure you can do the normal things.
